core/src/test/java/org/apache/iceberg/deletes/TestRoaringPositionBitmap.java

```diff
@@ -0,0 +1,323 @@
+import java.nio.ByteOrder;
+import java.util.Set;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableSet;
+import org.apache.iceberg.relocated.com.google.common.io.Resources;
+import org.junit.jupiter.api.Test;
+
+public class TestRoaringPositionBitmap {
+
+  private static final long BITMAP_OFFSET = 0xFFFFFFFFL + 1L;
+  private static final long CONTAINER_OFFSET = Character.MAX_VALUE + 1L;
+  private static final Set<String> SUPPORTED_OFFICIAL_EXAMPLE_FILES =
+      ImmutableSet.of("64map32bitvals.bin", "64mapempty.bin", "64mapspreadvals.bin");
+
+  @Test
+  public void testAdd() {
+    RoaringPositionBitmap bitmap = new RoaringPositionBitmap();
+
+    bitmap.add(10L);
+    assertThat(bitmap.contains(10L)).isTrue();
+
+    bitmap.add(0L);
+    assertThat(bitmap.contains(0L)).isTrue();
+
+    bitmap.add(10L);
+    assertThat(bitmap.contains(10L)).isTrue();
+  }
+
+  @Test
+  public void testAddPositionsRequiringMultipleBitmaps() {
+    RoaringPositionBitmap bitmap = new RoaringPositionBitmap();
+
+    // construct positions that differ in their high 32-bit parts (i.e. keys)
+    long pos1 = ((long) 0 << 32) | 10L; // high = 0, low = 10
+    long pos2 = ((long) 1 << 32) | 20L; // high = 1, low = 20
+    long pos3 = ((long) 2 << 32) | 30L; // high = 2, low = 30
+    long pos4 = ((long) 100 << 32) | 40L; // high = 1000, low = 40
+
+    bitmap.add(pos1);
+    bitmap.add(pos2);
+    bitmap.add(pos3);
+    bitmap.add(pos4);
+
+    assertThat(bitmap.contains(pos1)).isTrue();
+    assertThat(bitmap.contains(pos2)).isTrue();
+    assertThat(bitmap.contains(pos3)).isTrue();
+    assertThat(bitmap.contains(pos4)).isTrue();
+
+    assertThat(bitmap.cardinality()).isEqualTo(4);
+
+    assertThat(bitmap.serializedSizeInBytes()).isGreaterThan(4);
+  }
+
+  @Test
+  public void testAddRange() {
+    RoaringPositionBitmap bitmap = new RoaringPositionBitmap();
+
+    long startInclusive = 10L;
+    long endExclusive = 20L;
+    bitmap.addRange(startInclusive, endExclusive);
+
+    // assert that all positions in the range [10, 20) are added
+    for (long pos = startInclusive; pos < endExclusive; pos++) {
+      assertThat(bitmap.contains(pos)).isTrue();
+    }
+
+    // assert that positions outside the range are not present
+    assertThat(bitmap.contains(9L)).isFalse();
+    assertThat(bitmap.contains(20L)).isFalse();
+
+    // assert that the cardinality is correct (10 positions in range [10, 20))
+    assertThat(bitmap.cardinality()).isEqualTo(10);
+  }
+
+  @Test
+  public void testAddEmptyRange() {
+    RoaringPositionBitmap bitmap = new RoaringPositionBitmap();
+    bitmap.addRange(10, 10);
+    assertThat(bitmap.isEmpty()).isTrue();
+  }
+
+  @Test
+  public void testOr() {
+    RoaringPositionBitmap bitmap1 = new RoaringPositionBitmap();
+    bitmap1.add(10L);
+    bitmap1.add(20L);
+
+    RoaringPositionBitmap bitmap2 = new RoaringPositionBitmap();
+    bitmap2.add(30L);
+    bitmap2.add(40L);
+
+    bitmap1.or(bitmap2);
```

Comment:
Is there a better name for this method because it modifies `bitmap1`? And should we also validate that `bitmap2` is unmodified in this test?

I would probably prefer `addAll` or `setAllFrom` for the method name. The name `or` made me think that this would return a new union bitmap.

Reply:
I did follow the the naming and semantics of the Roaring library, but let me think about this one.